src/main/context-handler.ts

```diff
@@ -1,76 +1,42 @@
-import { CoreV1Api, KubeConfig } from "@kubernetes/client-node"
-import { ServerOptions } from "http-proxy"
-import * as url from "url"
+import type { PrometheusProvider, PrometheusService } from "./prometheus/provider-registry"
+import type { ClusterPreferences } from "../common/cluster-store";
+import type { Cluster } from "./cluster"
+import type httpProxy from "http-proxy"
+import url, { UrlWithStringQuery } from "url";
+import { CoreV1Api } from "@kubernetes/client-node"
+import { prometheusProviders } from "../common/prometheus-providers"
 import logger from "./logger"
 import { getFreePort } from "./port"
 import { KubeAuthProxy } from "./kube-auth-proxy"
-import { Cluster, ClusterPreferences } from "./cluster"
-import { prometheusProviders } from "../common/prometheus-providers"
-import { PrometheusService, PrometheusProvider } from "./prometheus/provider-registry"
 
 export class ContextHandler {
-  public contextName: string
-  public id: string
-  public url: string
-  public clusterUrl: url.UrlWithStringQuery
-  public proxyServer: KubeAuthProxy
-  public proxyPort: number
-  public certData: string
-  public authCertData: string
-  public cluster: Cluster
-
-  protected apiTarget: ServerOptions
-  protected proxyTarget: ServerOptions
-  protected clientCert: string
-  protected clientKey: string
-  protected secureApiConnection = true
-  protected defaultNamespace: string
-  protected kubernetesApi: string
+  public proxyPort: number;
+  public clusterUrl: UrlWithStringQuery;
+  protected kubeAuthProxy: KubeAuthProxy
+  protected apiTarget: httpProxy.ServerOptions
   protected prometheusProvider: string
   protected prometheusPath: string
-  protected clusterName: string
 
-  constructor(kc: KubeConfig, cluster: Cluster) {
-    this.id = cluster.id
-
-    this.cluster = cluster
-    this.clusterUrl = url.parse(cluster.apiUrl)
-    this.contextName = cluster.contextName;
-    this.defaultNamespace = kc.getContextObject(cluster.contextName).namespace
-    this.url = `http://${this.id}.localhost:${cluster.port}/`
-    this.kubernetesApi = `http://127.0.0.1:${cluster.port}/${this.id}`
```

Comment:
How these changes are related to Vue->React refactoring (I mean most changes in this file)? I'm a bit afraid that these will cause troubles without more testing.

Reply:
Vue-React + stores + overall design requires this changes..